codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerServiceGeneratorV2.kt

```diff
@@ -469,8 +472,131 @@ class ServerServiceGeneratorV2(
+            /// let handler = #{SmithyHttpServer}::routing::LambdaHandler::new(app);
+            /// lambda_http::run(handler).await.unwrap();
+            /// ```
+            ///
+            /// ## Building the $serviceName
+            ///
+            /// To construct [`$serviceName`] we use [`$builderName`] returned by [`$serviceName::builder_without_plugins`]
+            /// or [`$serviceName::builder_with_plugins`].
+            ///
+            /// #### Plugins
+            ///
+            /// The [`$serviceName::builder_with_plugins`] method, returning [`$builderName`], we saw earlier,
+            /// accepts a `plugin`. Plugins allow you to build middleware which is aware of the operation it is being applied to.
+            ///
+            /// ```rust,no_run
+            /// ## use #{SmithyHttpServer}::plugin::IdentityPlugin as LoggingPlugin;
+            /// ## use #{SmithyHttpServer}::plugin::IdentityPlugin as MetricsPlugin;
+            /// let plugins = PluginPipeline::new()
+            ///         .push(LoggingPlugin)
+            ///         .push(MetricsPlugin);
+            /// let builder = $serviceName::builder_with_plugins(plugins);
+            /// ```
+            ///
+            /// To know more about plugins, see: [`#{SmithyHttpServer}::plugin`].
+            ///
+            /// #### Handlers
+            ///
+            /// For each operation it has an associated setters accepting an async function conforming to the Smithy model.
+            /// The async functions, or "handlers", contain the business logic of your application.
+            ///
+            /// Every handler must take an `Input`, and optional [`extractor arguments`](#{SmithyHttpServer}::request), while return:
+            ///
+            /// * A `Result<Output, Error>` if your operation has modeled errors, or
+            /// * An `Output` otherwise.
+            ///
+            /// ```rust,ignore
+            /// async fn handler_fallible(input: Input, extensions: #{SmithyHttpServer}::Extension<T>) -> Result<Output, Error> { todo!() }
+            /// async fn handler_infallible(input: Input, extensions: #{SmithyHttpServer}::Extension<T>) -> Output { todo!() }
+            /// ```
+            ///
+            /// Handlers accept up to 8 extractors:
+            ///
+            /// ```rust,ignore
+            /// async fn handler_with_no_extensions(input: Input) -> ... { todo!() }
+            /// async fn handler_with_one_extension(input: Input, ext: #{SmithyHttpServer}::Extension<T>) -> ... { todo!() }
+            /// async fn handler_with_two_extensions(input: Input, ext0: #{SmithyHttpServer}::Extension<T>, ext1: #{SmithyHttpServer}::Extension<T>) -> ... { todo!() }
+            /// ...
+            /// ```
+            ///
+            /// #### Build
+            ///
+            /// When you have set all your operations, you can construct [`$serviceName`] using:
```

Comment:
We need to reword this, since you don't want to set all operations if you are going for `build_unchecked`.